src/main/java/com/hierynomus/smbj/share/DiskShare.java

```diff
@@ -466,6 +468,127 @@ private boolean exists(String path, EnumSet<SMB2CreateOptions> createOptions) th
         }
     }
 
+    /***
+     * Rename(Move) Directory/File. Default will using the false for replaceIfExist option
+     * @param path the path to the original directory/file
+     * @param newFileName the new directory/file name/path after rename
+     * @throws TransportException
+     * @throws SMBApiException
+     */
+    public void rename(String path, String newFileName) throws TransportException, SMBApiException {
```

Comment:
Ok, The File/Disk APIs are still hardly designed unfortunately, but what I would prefer is to have a `rename` (and/or `mv`) method on both `Disk` and `File` which checks whether the Entry was opened with the correct access level, and executes the command. Failing if the `DELETE` access has not been given in the `getFile`, `openDirectory` or `openFile` call.

API-wise it is nicer to have them on those domain objects, at least that is my opinion :)

And yes maybe we should move some more methods also while we're at it. I've been meaning to refactor that part anyway.